### PR TITLE
optimize: add git infos to jars

### PR DESCRIPTION
**Pull request by wangliang181230**
- [x] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
optimize: add git infos to jars.
添加git信息到jar包中。

图片

### CODE REVIEW

**Review by xingfudeshi — APPROVED** [review]
LGTM

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,9 +32,9 @@ jobs:
       - name: "Build with Maven"
         run: |
           if [ "${{ matrix.java }}" == "8" ]; then
-            ./mvnw -T 4C clean test -Dcheckstyle.skip=false -Dlicense.skip=false -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+            ./mvnw -T 4C clean test -Dcheckstyle.skip=false -Dlicense.skip=false -Dmaven.git-commit-id.skip=true -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
           elif [ "${{ matrix.java }}" == "17" ]; then
-            ./mvnw -T 4C clean test -Dcheckstyle.skip=true  -Dlicense.skip=true  -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
+            ./mvnw -T 4C clean test -Dcheckstyle.skip=true  -Dlicense.skip=true  -Dmaven.git-commit-id.skip=true -e -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn;
```
Comment (wangliang181230) [inline .github/workflows/build.yml:37]: git-commit-id-plugin，有时候会消耗很长时间，为了不影响测试效率，在所有测试的ci中，跳过该插件。

File: build/pom.xml [hunk 2]
```diff
@@ -80,10 +80,6 @@
         <junit-jupiter.version>5.8.2</junit-jupiter.version>
         <junit-platform.version>1.8.2</junit-platform.version>
 
-        <!-- For test -->
-        <junit-jupiter.version>5.8.2</junit-jupiter.version>
-        <junit-platform.version>1.8.2</junit-platform.version>
```
Comment (wangliang181230) [inline build/pom.xml:85]: 重复的内容，移除掉。

**Comment by wangliang181230** [comment]
@xingfudeshi I changed the base branch to `2.x`, PTAL.

File: .github/workflows/build.yml (hunk 1 above)
Comment (slievrly) [reply .github/workflows/build.yml:37]: just keep only necessary information

**Review by funky-eyes — APPROVED** [review]
LGTM
补充下change log
Please supplement the change log.

File: changes/en-us/2.x.md
Comment (funky-eyes) [inline changes/en-us/2.x.md:1]: I didn't add the author info.

File: build/pom.xml [hunk 3]
```diff
@@ -204,6 +201,25 @@
...
+                            <goals>
+                                <goal>revision</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                    <configuration>
+                        <verbose>true</verbose>
+                        <dateFormat>yyyy-MM-dd'T'HH:mm:ssZ</dateFormat>
+                        <generateGitPropertiesFile>true</generateGitPropertiesFile>
+                        <generateGitPropertiesFilename>${project.build.outputDirectory}/seata-git.properties</generateGitPropertiesFilename>
```
Comment (slievrly) [inline build/pom.xml:221]: includeOnlyProperties
^git.dirty$
^git.commit.time$
^git.build.version$
^git.build.time$
^git.branch$
^git.commit.id$

Comment (wangliang181230) [reply build/pom.xml:220]: Let me see if it's possible to configure it.

Comment (wangliang181230) [reply build/pom.xml:221]: Done, PTAL

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**